src/ctap2.rs

```diff
@@ -393,21 +388,12 @@ impl<UP: UserPresence, T: TrussedRequirements> Authenticator for crate::Authenti
             self.delete_resident_key_by_user_id(&rp_id_hash, &credential.user.id)
                 .ok();
 
-            let mut key_store_full = false;
-
-            // then check the maximum number of RK credentials
-            if let Some(max_count) = self.config.max_resident_credential_count {
-                let mut cm = credential_management::CredentialManagement::new(self);
-                let metadata = cm.get_creds_metadata();
-                let count = metadata
-                    .existing_resident_credentials_count
-                    .unwrap_or(max_count);
-                debug!("resident cred count: {} (max: {})", count, max_count);
-                if count >= max_count {
-                    error!("maximum resident credential count reached");
-                    key_store_full = true;
-                }
-            }
+            let mut key_store_full = !self.can_fit(serialized_credential.len())
+                || CredentialManagement::new(self).count_credentials()
+                    >= self
+                        .config
+                        .max_resident_credential_count
+                        .unwrap_or(MAX_RESIDENT_CREDENTIALS_GUESSTIMATE);
```

Comment:
As we now always apply a hard limit, it would make sense to make the `max_resident_credential_count` configuration required and remove the guesstimate constant.  This would also make the code easier to read.